stpsf/roman.py

```diff
@@ -633,31 +692,41 @@ def __init__(self):
         # https://roman.ipac.caltech.edu/sims/Param_db.html
         pixelscale = 110e-3  # arcsec/px
 
-        # Initialize the aberrations for super().__init__
+        # Initialize the aberrations for parent SpaceTelescopeInstrument class
         self._aberration_files = {}
         self._is_custom_aberration = False
         self._current_aberration_file = ''
 
         super().__init__('WFI', pixelscale=pixelscale)
 
         # Initialize the pupil controller
-        self._pupil_controller = WFIPupilController()
-        self._pupil_controller.set_base_path(self._datapath)
+        self._pupil_controller = WFIPupilController(self._datapath)
 
-        self.pupil_mask_list = list(self._pupil_controller.pupil_file_formatters.keys())
+        # self.pupil_mask_list = list(self._pupil_controller.pupil_file_formatters.keys())
+        self.pupil_mask_list = [fltr for fltr in self.filter_list.copy()
+                                if not fltr.startswith('GRISM')]
+        self.pupil_mask_list.append(np.str_('GRISM'))  # GRISM0/1 share pupils
 
         # Define default aberration files for WFI modes
-        self._aberration_files = {
-            'imaging': os.path.join(self._datapath, 'wim_zernikes_cycle9.csv'),
-            'prism': os.path.join(self._datapath, 'wsm_prism_zernikes_cycle9.csv'),
-            'grism': os.path.join(self._datapath, 'wsm_grism_zernikes_cycle9.csv'),
-            'custom': None,
-        }
+        # self._aberration_files = {
+        #     'imaging': os.path.join(self._datapath, 'wim_zernikes_cycle9.csv'),
+        #     'prism': os.path.join(self._datapath, 'wsm_prism_zernikes_cycle9.csv'),
+        #     'grism': os.path.join(self._datapath, 'wsm_grism_zernikes_cycle9.csv'),
+        #     'custom': None,
+        # }
+        self._aberration_files = {'custom': None}
+        self._aberration_files.update({
+            fltr: os.path.join(
+                self._datapath,
+                (f"{'WIM' if fltr.startswith(('F', 'W')) else 'WSM'}"
```

Comment:
Now that W146 was renamed to F146, we can remove the `starstwith("W")` part of this line.